hamiltonian_flow/models/harmonic_oscillator.py

```diff
@@ -0,0 +1,249 @@
+from functools import cached_property
+from typing import Dict, Literal, Optional, Union
+
+import numpy as np
+import pandas as pd
+from pydantic import BaseModel, computed_field
+
+
+class HarmonicOscillatorSystem(BaseModel):
+    """The params for the harmonic oscillator
+
+    :param omega: angular frequency of the harmonic oscillator
+    :param amplitude: the amplitude of the oscillation
+    :param phi: initial phase
+    """
+
+    omega: float
+    zeta: float = 0.0
+
+    @computed_field  # type: ignore[misc]
+    @cached_property
+    def period(self) -> float:
+        """period of the oscillator"""
+        return 2 * np.pi / self.omega
+
+    @computed_field  # type: ignore[misc]
+    @cached_property
+    def frequency(self) -> float:
+        """frequency of the oscillator"""
+        return 1 / self.period
+
+    @computed_field  # type: ignore[misc]
+    @cached_property
+    def type(
+        self,
+    ) -> Literal["simple", "under_damped", "critical_damped", "over_damped"]:
+        """which type of harmonic oscillators"""
+        if self.zeta == 0:
+            return "simple"
+        elif self.zeta < 1:
+            return "under_damped"
+        elif self.zeta == 1:
+            return "critical_damped"
+        else:
+            return "over_damped"
+
+
+class HarmonicOscillatorIC(BaseModel):
+    """The initial condition for a harmonic oscillator
+
+    :param x0: the initial displacement
+    :param v0: the initial velocity
+    """
+
+    x0: float = 1.0
+    v0: float = 0.0
+
+
+class HarmonicOscillator:
+    r"""Generate time series data for a [harmonic oscillator](https://en.wikipedia.org/wiki/Harmonic_oscillator).
```

Comment:
Would it be a better practice if we leave the lengthy docstring outside the python script?

Reply:
> Would it be a better practice if we leave the lengthy docstring outside the python script?

Such as?

Reply:
I do not have a clear idea, but can we make a separate `yaml` file or something similar?

Reply:
> I do not have a clear idea, but can we make a separate `yaml` file or something similar?

I love detailed docstrings. I don't see any drawbacks yet.

We can move these details to mkdocs but that also means much more work. I would say let's keep this until we run in to problems.